javaparser-core/src/main/java/com/github/javaparser/JavaParser.java

```diff
@@ -114,6 +132,18 @@ private GeneratedJavaParser getParserForProvider(Provider provider) {
         assertNotNull(start);
         assertNotNull(provider);
 
+        // FIXME: Surely there's a better way...?
+        String x = providerToString(provider);
+        // close the old provider then open a new one...
+        try {
+            provider.close();
```

Comment:
Surely there's a better way to achieve this than consuming the whole stream, interpreting it, and then forwarding a brand new stream (`provider`) based on this new string... 

Note that the original stream will be unavailable due to having been fully consumed.

Reply:
I think you want to create a preprocessor for this, like the unicode escaping preprocessor. Then add a configuration option so those who don't care about this feature can turn it off :-)

Reply:
I didn't really understand the unicode preprocessor(!)

What I could determine is that it modifies the stream, but isn't able to store any state or pass any results on..? Or maybe I misunderstood.

The other aspect (I can't remember if I mentioned this in another comment) is that during the pre-processing stage the processors don't seem to have access to either the config or the the resulting nodes.

Presumably it's inappropriate to store this on the processor itself... 

If that's something you can give guidance on, then that would be great! Otherwise I'll ruminate and think on this some more :)

Reply:
I've done removed this and replaced it with a preprocessor - marking as "resolved".